massa-execution-exports/src/mapping_grpc.rs

```diff
@@ -49,15 +51,53 @@ pub fn to_querystate_filter(
                 ExecutionQueryRequestItem::AddressBytecodeFinal(Address::from_str(&value.address)?),
             ),
             exec::RequestItem::AddressDatastoreKeysCandidate(value) => {
+                // take the minimum of the limit and the max_datastore_query_config if it is set
+                let count = value
+                    .limit
+                    .map(|c| {
+                        max_datastore_query_config
+                            .map(|conf| c.min(conf))
```

Comment:
same comment here: fail if the user sent a too high limit instead of silently capping

Reply:
done